core/src/main/java/org/axonframework/messaging/Message.java

```diff
@@ -91,4 +93,28 @@
      * @return a copy of this message with the given MetaData
      */
     Message<T> andMetaData(Map<String, ?> metaData);
+
+    /**
+     * Serialize the payload of this message using given {@code serializer}, using given
+     * {@code expectedRepresentation}. This method <em>should</em> return the same SerializedObject instance when
+     * invoked multiple times using the same serializer.
+     *
+     * @param serializer             The serializer to serialize payload with
+     * @param expectedRepresentation The type of data to serialize to
+     * @param <R>                    The type of data to serialize to
```

Comment:
Nit: I'd rephrase this to be a little different from `@param expectedRepresentation`, something like:
`@param <R> the type of the serialized data`